### PR TITLE
Feature/to npm package

### PR DESCRIPTION
**Pull request by juliencrn**
# Goal
Create a NPM packages to use the hooks as a library

# Features
- Split the codebase into one repo for the static website and one other for the hooks
- Add Contributors to the README
- Share config (Jest, Prettier, eslint, Typescript) between packages
- Add lint-staged and husky to fail earlier

Big thanks to @vfonic, @bclermont and @strange2x who convinced me to begin this journey.

PS: My bad, it was my first try with a mono-repo, the commit history is dirty, I've tried a lot of things. Apologies! 🙏 

If someone has any feedback, it's really welcome 👍 

Related issue: #26

### CODE REVIEW

**Comment by juliencrn** [comment]
Notes: The netlify deploy fail is normal. The current deploy settings are configured for the master branch with Gatsby at the root of the repo. Now, we use lerna at the root and the Gatsby website is in `packages/frontend`.

**Comment by strange2x** [comment]
Nice work man.. Thanks a lot for the effort..

File: README.md [hunk 1]
```diff
@@ -1,131 +1,165 @@
...
 ## 🤔 About
 
-This is the repository for [usehooks.ts](https://usehooks-typescript.com/), a [Gatsby](https://www.gatsbyjs.org) powered blog hosted with Github & netlify that publishes easy to understand React Hook code snippets.
+Initially, useHooks.ts was a [Gatsby](https://www.gatsbyjs.org) powered blog hosted with Github & netlify that publishes easy to understand React Hook code snippets.
 
-UseHooks.ts has been strongly inspired by [usehooks.com](https://usehooks.com), thanks to him. My objective here is to publish my hooks in Typescript in order to find them quickly
+But now, it's a monorepo containing:
+
+- A static website used as hooks documentation ([Link](https://usehooks-typescript.com/)).
+- A NPM packages containing the hooks library ([Link](https://www.npmjs.com/package/usehooks-ts)).
```
Comment (vfonic) [inline README.md:40]: *An (it's weird, but it's the sound, not the letter that counts, hence "An En-pi-em" vs "A En-pi-em" - NPM's first sounds is a vowel)
https://english.stackexchange.com/questions/105116/is-it-a-user-or-an-user

*package

**Review by vfonic — APPROVED** [review]
I did not know this is so much work! 😱 

Thank you so much for pulling this off! 🎉 

Feel free to ignore my one grammar related comment. 😁

File: README.md [hunk 2]
```diff
@@ -1,131 +1,165 @@
...
+- [`useScript()`](https://usehooks-typescript.com/react-hook/use-script)
+- [`useSsr()`](https://usehooks-typescript.com/react-hook/use-ssr)
+- [`useTimeout()`](https://usehooks-typescript.com/react-hook/use-timeout)
+- [`useWindowSize()`](https://usehooks-typescript.com/react-hook/use-window-size)
+
+<!-- HOOKS:END -->
 
 ## 👉 Installation
 
 **Note**: The project use `Node@14` and `npm@6`.
```
Comment (vfonic) [inline README.md:76]: uses*

File: README.md (hunk 1 above)
Comment (juliencrn) [reply README.md:40]: I also asked myself the question haha. 

Thanks for the feedback.

**Pull request merged**